app/scripts/controllers/app-state.js

```diff
@@ -40,20 +43,36 @@ class AppStateController extends EventEmitter {
    * Get a Promise that resolves when the extension is unlocked.
    * This Promise will never reject.
    *
+   * @param {boolean} showUnlockRequest - Whether the extension notification popup should be opened.
    * @returns {Promise<void>} A promise that resolves when the extension is
    * unlocked, or immediately if the extension is already unlocked.
    */
-  getUnlockPromise () {
+  getUnlockPromise (showUnlockRequest) {
```

Comment:
Nit: maybe this could be renamed to `shouldShowUnlockRequest`? Since the function is also called `showUnlockRequest`, I thought this _was_ the function on my first read through until reading that JSDoc comment more closely.

Reply:
Or, alternatively, you could remove this parameter. It looks like it's always set to `true` in practice.

Reply:
Good point. I renamed it. It is always set to `true` in practice, but in case we ever want to use it for something else, it ought to save someone else time in the future, at virtually no cost today.

Reply:
I'm having trouble thinking of a situation where we'd want it to be `false` though. In general, keeping a parameter in "just in case" seems like an odd choice, but 🤷‍♂️